db/migrations/0018_add_lti_usage.py

```diff
@@ -0,0 +1,43 @@
+'''
+Migration for MiVideo media started hourly table
+'''
+
+from yoyo import step
+
+steps = [
+    step('''
+        CREATE TABLE IF NOT EXISTS lti_placement_types (
+            id INTEGER NOT NULL AUTO_INCREMENT,
+            description VARCHAR(100) NOT NULL,
+            table_name VARCHAR(64) NOT NULL,
+            PRIMARY KEY (id)
+        ) ENGINE = InnoDB DEFAULT CHARSET = utf8mb4;
+    '''),
+    step('''
+        INSERT IGNORE INTO lti_placement_types
+        (id, description, table_name) VALUES
+        (0, 'Zoom', 'lti_zoom_placement')
+    '''),
+    step('''
+        CREATE TABLE IF NOT EXISTS lti_placement (
+            id BIGINT NOT NULL AUTO_INCREMENT,
+            account_id INTEGER NOT NULL,
+            course_id INTEGER NOT NULL,
+            course_name VARCHAR(200),
```

Comment:
Yeah, I think we could toss `account_id` and `course_name` if it's a Canvas course ID that can be linked back to the Canvas dataset. Like with the MiVideo stuff, we can't enforce a relationship because records are dropped, but you can still join.

Reply:
Yes, if we had a way to populate this course table easier I'd have done this but I figured it didn't need to be part of this change. The only place I saw it populated was in inventory.py on or near line 365.

A method to actually populate to the course tables doesn't exist as part of this code yet independent of this and I didn't see where/if MiVideo was doing this differently. I would actually want to call some utility method that would create the course if it doesn't exist, then be able to link to that.

Reply:
I understand your thinking here, but I don't know if we want multiple processes to be responsible for creating course records. We did talk briefly in the past about the possibility of making your script dependent on `course_inventory`, i.e. fetch the course data from the database rather than running a separate process; then you'd know that the `canvas_id` really was in `course`, even if we couldn't enforce it.

Reply:
Sure but that's the way it is now. It's also possible that this may pull down different courses than the course_inventory since course_inventory has some other criteria and this will eventually just pull from the report.

We could refactor this whole thing and have

A top level job that processes though courses
- Get courses in a term that match the criteria
  - Runs the course inventory
- Get courses from the LTI placement report
  - Runs the LTI placements

Or we could just have some method that creates courses for all of these scripts. I don't know how MiVideo process would fit into this courses job?

Reply:
I think we should just create a new issue to deal with the course FK problem as it's going to be bigger than getting this data into the database. 

I agree that multiple processes shouldn't be creating this but I don't know how a single process can easily accommodate for everything that may need to use this without having a pre-process separate from either of these before to update the courses (without dropping it).

Reply:
Sounds fine to me. Do you want to create the issue or do you want me to?

Reply:
I'll create this